### PR TITLE
Add cci org shell command

### PR DESCRIPTION
**Pull request by davidmreed**
# Critical Changes

# Changes

 - Add the `cci org shell` command, which opens a Python shell pre-populated with a `simple_salesforce` session on the selected org (as `sf`).

# Issues Closed

### CODE REVIEW

**Comment by davidmreed** [comment]
I prefer (2). In `cumulusci.utils`, would you say?

**Comment by davisagli** [comment]
@davidmreed how about a new utils.py or rest.py module in cumulusci.salesforce_api. I always feel bad adding more things to cumulusci.utils.

File: cumulusci/cli/cci.py [hunk 1]
```diff
@@ -1009,6 +1012,27 @@ def org_scratch_delete(config, org_name):
...
+@pass_config
+def org_shell(config, org_name):
+    org_name, org_config = config.get_org(org_name)
+    org_config.refresh_oauth_token(config.keychain)
+
+    sf = get_simple_salesforce_connection(config.project_config, org_config)
+
+    code.interact(
+        banner="Use `sf` to access org `{}` via simple_salesforce".format(org_name),
+        local=dict(globals(), **locals()),
```
Comment (davisagli) [inline cumulusci/cli/cci.py]: Why include the `cci` module's globals here?

Should we include a direct reference to `project_config` too?

Comment (davidmreed) [reply cumulusci/cli/cci.py]: I assumed (without a good reason!) that there was a design decision behind doing so in the `shell` command that would also apply here - but if that command does so more with an eye towards facilitating CCI debugging, there's no need to do so here, and exposing just `sf` and `project_config` (perhaps also a `Salesforce_Bulk` instance) would be enough.

Comment (davisagli) [reply cumulusci/cli/cci.py]: I'd suggest exposing org_config too.

Comment (davidmreed) [reply cumulusci/cli/cci.py]: Done!

**Pull request merged**